scst-store/amr/graphql-query.hbs.md

```diff
@@ -0,0 +1,262 @@
+                              {key: "region", value: "east1b"} ]
+    })
+    ```
+
+### <a id='location-fields'></a> Locations Fields
+
+Below are fields that users can choose to return in the GraphQL query. At least one field needs to be specified.
+
+- `alias`: human-readable identifier of the location
+- `reference`: string UID representing the location
+- `labels`: labels associated with the location, has the following fields
+  - `key`: key of the label
+  - `value`: value of the label
+  
+### <a id='sample-locations-query'></a> Sample Locations Queries
+
+- Query for location with the `alias` `"test-cluster"`. 
+  In the query results, return information on the `reference`, `alias`, and `labels` showing both `key` and `value`.
+  ```
+  query getLocationByAlias {
+    locations(query:{alias: "test-cluster"}) {
+      reference
+      alias
+      labels {
+        key
+        value
+      }
+    }
+  }
+  ```
+  
+- Query for locations that contains two `labels`, one with `key` `"env"` and one `{key: "region", value: "east1b"}`.
+  In the query results, return information on `alias` and `reference` only.
+  ```
+  query getLocationByLabel {
+    locations(query:{labels:[{key: "env"}, 
+                              {key: "region", value: "east1b"} ]
+    }) {
+      reference
+      alias
+    }
+  }
+  ```
+
+## <a id='query-apps'></a> Querying for Apps
+
+This section provides details on the GraphQL query arguments, and the list of fields available for `Apps`.
+
+### <a id='apps-query-args'></a> Apps Query Arguments
+
+Below are the supported arguments that users can specify when querying for `Apps`. They are all optional.
+
+- `latest`: When set to true, it returns the latest result from the given query. When set to false, it returns all results from the given query. 
+  When not specified, it defaults to false. This field is a `Boolean` value.
+  
+  Example:
+  ```
+  apps(latest: true)
+  ```
+
+- `query`: expects an object that specifies additional arguments used to query, supported arguments in this query object are listed below
+  
+  - `name`: the name of the app, `String` value
+    
+    Example:
+    ```
+    apps(query: {name: "my-app"})
+    ```
+    
+  - `namespace`: the namespace where the app is deployed, `String` value
+    
+    Example:
+    ```
+    apps(query: {namespace: "dev-namespace"})
+    ```
+    
+  - `location`: the location where the app is deployed, expects a location query object as described in the section above (see [Location Query Arguments](#location-query-args))
+
+    >**Note** Currently `alias` is the only supported field for `location` when querying for `apps`. The other location fields are available in the schema but have no effect.
+
+    Example:
+    ```
+    apps(query: {location: {alias: "test-cluster"}})
+    ```
+    
+  - `order`: order results by a specified `field` in a given `direction`, expects a key-value list to specify the `field` and `direction`. 
```

Comment:
❓ I can't remember the implementation but does field default to timestamp if it's not specified? Or does it return 400 Bad Request? If it's required, let's replace `expects a key-value list` to `requires a key-value list`.

Reply:
It will return an error saying GraphQL validation failed because it's expecting a field. Personally `expects` and `requires` seems very similar in meaning, but if you feel strongly about the wording I can make a new MR to docs-tap to change this word. Because the original MR got merged in already 😅